### PR TITLE
linked time: Implement getHighestStep and getLowestStep

### PR DESCRIPTION
**Pull request by JamesHollyer**
* Motivation for features / changes
This is the first step in implementing the functions needed to add fob dragging to the Scalar card. 

* Technical description of changes
There is no guarantee that the minMax object is in order to we simply check which member is greater than the other and return the appropriate one.

### CODE REVIEW

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts [hunk 1]
```diff
@@ -86,4 +86,41 @@ describe('ScalarLinkedTimeFobController', () => {
...
+    it('gets the highest step when minMax is not in order', () => {
+      const minMax: [number, number] = [2, 0];
+      const fixture = createComponent({minMax});
+
+      const highestStep = fixture.componentInstance.getHighestStep();
+
+      expect(highestStep).toBe(2);
+    });
+  });
+  describe('getLowestStep', () => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts:108]: nit: newline between each "describe" and "it"

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts [hunk 2]
```diff
@@ -41,11 +41,13 @@ export class ScalarCardLinkedTimeFobController implements FobCardAdapter {
 
   // TODO(jameshollyer): Implements remaining methods for scalar card fob controller.
   getHighestStep(): number {
-    return -1;
+    let minMax = this.minMax;
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts:44]: nit: const

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts [hunk 3]
```diff
@@ -41,11 +41,13 @@ export class ScalarCardLinkedTimeFobController implements FobCardAdapter {
...
+    return minMax[0] < minMax[1] ? minMax[1] : minMax[0];
   }
 
   getLowestStep(): number {
-    return -1;
+    let minMax = this.minMax;
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts:49]: nit: const

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts [hunk 4]
```diff
@@ -86,4 +86,41 @@ describe('ScalarLinkedTimeFobController', () => {
...
+  describe('getHighestStep', () => {
+    it('gets the highest step when minMax is in order', () => {
+      const minMax: [number, number] = [0, 2];
+      const fixture = createComponent({minMax});
+
+      const highestStep = fixture.componentInstance.getHighestStep();
+
+      expect(highestStep).toBe(2);
+    });
+    it('gets the highest step when minMax is not in order', () => {
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts:99]: ditto

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts [hunk 5]
```diff
@@ -86,4 +86,41 @@ describe('ScalarLinkedTimeFobController', () => {
       );
     });
   });
+
+  describe('getHighestStep', () => {
+    it('gets the highest step when minMax is in order', () => {
+      const minMax: [number, number] = [0, 2];
+      const fixture = createComponent({minMax});
```
Comment (japie1235813) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts:93]: minMax variable is not reused. this two lines could be combined.
const fixture = createComponent({minMax: [0, 2]});

and the same for all other test cases

**Review by japie1235813 — COMMENTED** [review]
LGTM once all nit resolved. I didn't leave comment at every places for the same issue. Please check carefully.

**Review by japie1235813 — APPROVED** [review]

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts (hunk 2 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts]: Done.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts (hunk 3 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_controller.ts]: done.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts (hunk 5 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts]: Makes sense to me. Changed.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts (hunk 4 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts:99]: new line added.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts (hunk 1 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_linked_time_fob_test.ts:108]: done.

**Review by JamesHollyer — COMMENTED** [review]
Pretty sure I got all the repeats. Thanks.

**Pull request merged**